contracts/actions/liquity/LiquityOpen.sol

```diff
@@ -0,0 +1,113 @@
+import "../ActionBase.sol";
+
+contract LiquityOpen is ActionBase, LiquityHelper {
+    using TokenUtils for address;
+
+    /// @inheritdoc ActionBase
+    function executeAction(
+        bytes[] memory _callData,
+        bytes[] memory _subData,
+        uint8[] memory _paramMapping,
+        bytes32[] memory _returnValues
+    ) public payable virtual override returns (bytes32) {
+        (
+            uint256 maxFeePercentage,
+            uint256 collAmount,
+            uint256 LUSDAmount,
+            address from,
+            address to,
+            address upperHint,
+            address lowerHint
+        )= parseInputs(_callData);
+
+        maxFeePercentage = _parseParamUint(maxFeePercentage, _paramMapping[0], _subData, _returnValues);
+        collAmount = _parseParamUint(collAmount, _paramMapping[1], _subData, _returnValues);
+        LUSDAmount = _parseParamUint(LUSDAmount, _paramMapping[2], _subData, _returnValues);
+        from = _parseParamAddr(from, _paramMapping[3], _subData, _returnValues);
+        to = _parseParamAddr(from, _paramMapping[4], _subData, _returnValues);
+
+        uint256 troveOwner = _liquityOpen(maxFeePercentage, collAmount, LUSDAmount, from, to, upperHint, lowerHint);
+        return bytes32(troveOwner);
+    }
+
+    /// @inheritdoc ActionBase
+    function executeActionDirect(bytes[] memory _callData) public virtual payable override {
+        (
+            uint256 maxFeePercentage,
+            uint256 collAmount,
+            uint256 LUSDAmount,
+            address from,
+            address to,
+            address upperHint,
+            address lowerHint
+        )= parseInputs(_callData);
+
+        _liquityOpen(maxFeePercentage, collAmount, LUSDAmount, from, to, upperHint, lowerHint);
+    }
+
+    /// @inheritdoc ActionBase
+    function actionType() public pure virtual override returns (uint8) {
+        return uint8(ActionType.STANDARD_ACTION);
+    }
+
+    //////////////////////////// ACTION LOGIC ////////////////////////////
+
+    /// @notice Opens up an empty trove
+    function _liquityOpen(
+        uint256 _maxFeePercentage,
+        uint256 _collAmount,
+        uint256 _LUSDAmount,
+        address _from,
+        address _to,
+        address _upperHint,
+        address _lowerHint
+    ) internal returns (uint256) {
+        if (_collAmount == type(uint256).max) {
+            _collAmount = TokenUtils.WETH_ADDR.getBalance(_from);
+        }
+        WETH_ADDR.pullTokensIfNeeded(_from, _collAmount);
+        TokenUtils.withdrawWeth(_collAmount);
+
+        BorrowerOperations.openTrove{value: _collAmount}(_maxFeePercentage, _LUSDAmount, _upperHint, _lowerHint);
+
+        LUSDTokenAddr.withdrawTokens(_to, _LUSDAmount);
+
+        logger.Log(
+            address(this),
+            msg.sender,
+            "LiquityOpen",
+            abi.encode(_maxFeePercentage, _collAmount, _LUSDAmount, _from, _to)
+        );
+
+        return uint256(address(this));
```

Comment:
Converting the address to uint doesn't make sense, also address(this) is DSProxy so redundant information. Return collAmount at that might change if it's uint.max